File: frontend-web/news-app/src/app/shared/models/post.ts
import { Review } from './review';
import { Comment } from './comment';
export class Post {
    id: Number;
    title: string;
    content: string;
    author: string;
    date: Date;
    concept: boolean;
    approved: boolean;
    published: boolean;
    review: Review | null = null;
    comments: Comment[] = [];
  
    constructor(id:Number,title: string, content: string, author: string, date: Date, concept: boolean, approved: boolean, published: boolean, review: Review, comments: Comment[] = []) {
      this.title = title;
      this.content = content;
      this.author = author;
      this.date = date;
      this.concept = concept;
      this.approved = approved;
      this.published = published;
      this.review = review;
      this.comments = comments || [];
   
        
    
      this.id = id;
    }
  }

File: backend-java/post-service/src/main/java/pxl/be/post/domain/Post.java
package pxl.be.post.domain;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "post")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Getter
    private Long id;

    @Getter
    @Setter
    @ElementCollection
    @CollectionTable(name = "post_comment_ids", joinColumns = @JoinColumn(name = "post_id"))
    @Column(name = "comment_id")
    private List<Long> commentIds;

    @Getter
    @Setter

    private Long reviewId;
    @Getter
    @Setter
    private String title;

    @Getter
    @Setter
    private String content;

    @Getter
    @Setter
    private boolean isConcept = true;

    @Getter
    private boolean isApproved = false;
    @Getter
    private boolean isPublished = false;

    @Getter
    private String author;

    @Getter
    @Setter
    private Date date;


    public void setIsApproved(boolean isApproved) {
        setReviewId(null);
        this.isApproved = isApproved;
    }

    public void setContent(String content) {
        setIsApproved(false);
        this.content = content;
    }
    public void setConcept(boolean concept) {
        if(concept){
            setReviewId(null);
        }
        this.isConcept = concept;
    }

    public void setTitle(String title) {
        setIsApproved(false);
        this.title = title;
    }

    public void publish() {
        if(!isApproved){
            throw new IllegalStateException("Post is not approved");
        }else if(isConcept){
            throw new IllegalStateException("Post is a concept");
        }
        this.isPublished = true;


    }

    public void addComment(Long commentId) {
        commentIds.add(commentId);
    }
    public void removeComment(Long commentId) {
        commentIds.remove(commentId);
    }


}
